jpa/subsystem/src/main/java/org/jboss/as/jpa/beanmanager/PersistenceIntegrationWithCDI.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * PersistenceIntegrationWithCDI will setup Persistence/CDI integration as mentioned in jakarta.ee/specifications/platform/11/jakarta-platform-spec-11.0#a441
+ * <p>
+ * Obtaining an Entity Manager using CDI:
+ * A Jakarta EE container must feature built-integration of Jakarta Persistence with the CDI bean manager, allowing injection of a container-managed
+ * entity manager factory using the annotation jakarta.inject.Inject.
+ * <p>
+ * For each persistence unit, the container must make available a bean with:
+ * - bean type EntityManager,
+ * - the qualifiers specified by qualifier XML elements in persistence.xml, or jakarta.enterprise.inject.Default, if no qualifiers are explicitly specified,
+ * - the scope specified by the scope XML element in persistence.xml, or jakarta.transaction.TransactionScoped if no scope is explicitly specified,
+ * - no interceptor bindings,
+ * - a bean implementation which satisfies the requirements of this specification for a container-managed entity manager.
+ * <p>
+ * "EntityManager must also be available via the additional access methods specified at the beginning of this section."
+ * <p>
+ * Injecting an Entity Manager Factory using CDI:
+ * A Jakarta EE container must feature built-in integration of Jakarta Persistence with the CDI bean manager,
+ * allowing injection of a container-managed entity manager using the annotation jakarta.inject.Inject.
+ * <p>
+ * For each persistence unit, the container must make available a bean with:
+ * - bean type EntityManagerFactory,
+ * - the qualifiers specified by qualifier XML elements in persistence.xml, or jakarta.enterprise.inject.Default, if no qualifiers are explicitly specified,
+ * - scope jakarta.enterprise.context.ApplicationScoped,
+ * - bean name given by the name of the persistence unit,
+ * - no interceptor bindings,
+ * - a bean implementation which satisfies the requirements of the Persistence specification for a container-managed entity manager factory.
+ * <p>
+ * Furthermore, the container must make available five beans with:
+ * - bean types CriteriaBuilder, PersistenceUnitUtil, Cache, SchemaManager, and Metamodel, respectively,
+ * - the qualifiers specified by qualifier XML elements in persistence.xml, or jakarta.enterprise.inject.Default, if no qualifiers are explicitly specified,
+ * - scope jakarta.enterprise.context.Dependent,
+ * - no interceptor bindings,
+ * - a bean implementation which simply obtains the instance of the bean type by calling the appropriate getter method of the EntityManagerFactory bean.
+ * - To access these bean types (CriteriaBuilder, PersistenceUnitUtil, Cache, SchemaManager, and Metamodel) from callsites that use @Resource or JNDI lookup,
+ * users must first obtain the EntityManagerFactory and then use the appropriate getter methods.
+ *
+ * @author Scott Marlow
+ */
+public class PersistenceIntegrationWithCDI {
+
+    private static final List<String> defaultQualifier = new ArrayList<String>();
```

Comment:
This should be:

private static final List<String> defaultQualifier = List.of("jakarta.enterprise.inject.Default");

Gets rid of the static initializer block and results in an immutable collection.